app/controllers/WorkerController.php

```diff
@@ -14,38 +17,69 @@ class WorkerController extends ControllerBase
     private $workerService;
 
     /**
-     * Initialises the workerService property
+     * @var AuthenticationService
      */
-    public function onConstruct()
+    private $authenticationService;
+
+    /**
+     * @var WorkerValidator
+     */
+    private $workerValidator;
+
+    /**
+     * @var WorkerModel
+     */
+    private $workerModel;
+
+    /**
+     * Initialises the workerService, workerValidator and workerModel properties
+     * @return void
+     */
+    public function onConstruct(): void
     {
-        $this->workerService = $this->getDi()->get(WorkerService::class);
+        $getDI                 = $this->getDi();
+        $this->workerService   = $getDI->get(WorkerService::class);
+        $this->authenticationService   = $getDI->get(AuthenticationService::class);
+        $this->workerValidator = $getDI->get(WorkerValidator::class);
+        $this->workerModel     = $getDI->get(WorkerModel::class);
```

Comment:
wh ydo you need validations in controoler?

Reply:
I call the $this->workerValidator->validation($worker) to return an array instead of an object

Reply:
you can't have validations in controllers and services. validations are made in just 1 place